code/game/mecha/components/armor.dm

```diff
@@ -0,0 +1,238 @@
+
+/obj/item/mecha_parts/component/armor
+	name = "mecha plating"
+	icon = 'icons/mecha/mech_component.dmi'
+	icon_state = "armor"
+	w_class = ITEMSIZE_HUGE
+	origin_tech = list(TECH_DATA = 1, TECH_ENGINEERING = 2)
+
+	component_type = MECH_ARMOR
+
+	start_damaged = FALSE
+
+	emp_resistance = 4
+
+	required_type = null	// List, if it exists. Exosuits meant to use the component.
+
+	integrity_danger_mod = 0.4	// Multiplier for comparison to max_integrity before problems start.
+	max_integrity = 120
+
+	internal_damage_flag = MECHA_INT_TEMP_CONTROL
+
+	step_delay = 1
+
+	var/deflect_chance = 10
+	var/list/damage_absorption = list(
+									"brute"=0.8,
+									"fire"=1.2,
+									"bullet"=0.9,
```

Comment:
Great googly indentation, batman! (I don't think 8 extra indents are really necessary, but you should vertically align the numbers)

Reply:
This shows up as far more indented here on Github, it's in-line with the list( component on DM